packages/core/src/checkout-buttons/strategies/amazon-pay-v2/amazon-pay-v2-button-options.ts

```diff
@@ -1,6 +1,22 @@
+import { BuyNowCartRequestBody } from '../../../cart';
 import { AmazonPayV2ButtonParameters } from '../../../payment/strategies/amazon-pay-v2';
 
+export function isWithBuyNowFeatures(
+    options?: AmazonPayV2ButtonInitializeOptions,
+): options is WithBuyNowFeature {
+    return !!(options as WithBuyNowFeature)?.buyNowInitializeOptions;
+}
+
+export interface WithBuyNowFeature {
+    /**
+     * The options that are required to initialize Buy Now functionality.
+     */
+    buyNowInitializeOptions?: {
+        getBuyNowCartRequestBody?(): BuyNowCartRequestBody | void;
+    };
+}
+
 /**
- * The required config to render the AmazonPayV2 buttton.
+ * The required config to render the AmazonPayV2 button.
  */
-export type AmazonPayV2ButtonInitializeOptions = AmazonPayV2ButtonParameters;
+export type AmazonPayV2ButtonInitializeOptions = AmazonPayV2ButtonParameters | WithBuyNowFeature;
```

Comment:
Out of curiosity why don't we do the same as paypal, where buy now part becomes an optional in the options object. cc @serhii-tkachenko

Reply:
@animesh1987 it's not necessary. I think this message should answer of your question https://github.com/bigcommerce/checkout-sdk-js/pull/1742#discussion_r1065126812

Reply:
Thanks for the context 🙇